### PR TITLE
Issue/5281 consolidate price translation

### LINKED ISSUE #5281: Plans: consolidate price translation
It looks like we're just localising it differently: "%@ per year" on the plan details vs price + "per year" on the plans list

Also:
> the test `testPlanListRowAttributedTitleWhenNotCurrent()` is broken when the simulator isn't in English: `expected to equal <Title $99 per year>, got <Title $99 al año>`.

### PR DESCRIPTION
**Pull request by frosty**
Fixes #5281. `PlanListViewController` and `PlanDetailViewController`'s header view now use the same localised string to construct their price period information. I've also updated two of the plan list tests that failed if the simulator was set to a different language / locale.

To test: Ensure that plan list and plan detail screens show the same (correct) price info and that it gets localized correctly. Ensure the plan list view controller tests pass.

Needs review: @koke

### CODE REVIEW

**Comment by koke** [comment]
I see prices twice:

simulator screen shot 09 may 2016 17 14 18

Which reminds me I used that approach because the original designs had a different style for price and "per year".

I'd rather not go the HTML route again, and wonder if we're needing some sort of basic markup for translations?

**Comment by frosty** [comment]
Ugh, sorry – can't shouldn't have missed this. I'll take a look at another way of handling this. 

I'm not sure if implementing our own markup (and parsing of it) is a bit of overkill for the situations we currently need to handle? It might be possible to make the styling of HTML attributed strings cleaner and more reusable. I'll try a few things out.

**Comment by frosty** [comment]
So I've abstracted out the styled HTML attributed string approach that we've used in a couple of other places (to `NSAttributedString+StyledHTML`) and used it here to improve the way we're constructing this string. It now becomes:

```
let pricePeriod = String(format: NSLocalizedString("%@ <em>per year</em>", comment: "Plan yearly price"), price)

let attributes: StyledHTMLAttributes = [ .BodyAttribute: priceAttributes,
                                         .EmTagAttribute: pricePeriodAttributes ]

let attributedPricePeriod = NSAttributedString.attributedStringWithHTML(pricePeriod, attributes: attributes)
```

The original issue I was trying to fix here (#5281) is actually no longer valid – this string _has_ to be different due to the different formatting of the price information. This new approach means, however, that the entire string (price + period) is localized together so the order can be changed if necessary.

I've updated the few other places that were styling HTML strings to use the new extension. Note that the label in `Signin2FAViewController` will appear black (instead of white, as intended) until https://github.com/wordpress-mobile/WordPress-Shared-iOS/pull/97 is merged (and we update the shared pod) to correctly handle creating hex strings from grayscale colors.

Over to you, @koke!

File: WordPress/Classes/Extensions/NSAttributedString+StyledHTML.swift [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+extension NSAttributedString {
+    /// Creates an `NSAttributedString` with the styles defined in `attributes` applied.
+    /// - parameter htmlString: The string to be styled. This can contain HTML
+    ///    tags to markup sections of the text to style, but should not be wrapped
+    ///    with `<html>`, `<body>` or `<p>` tags. See `HTMLAttributeType` for supported tags.
+    /// - parameter attributes: A collection of style attributes to apply to `htmlString`.
+    ///    See `HTMLAttributeType` for supported attributes.
+    ///
+    ///    **Notes:**
```
Comment (koke) [inline WordPress/Classes/Extensions/NSAttributedString+StyledHTML.swift]: You can use `- note:` here

https://developer.apple.com/library/ios/documentation/Xcode/Reference/xcode_markup_formatting_ref/Note.html

**Comment by koke** [comment]
I like the new approach to markup. It's not ideal but at least the mess is contained. There's a big merge conflict since #5333 moved some of the changed stuff to other files. Can you resolve that before I do some final testing/review?

**Comment by frosty** [comment]
Conflict fixed!

**Comment by koke** [comment]
> Note that the label in Signin2FAViewController will appear black

It also appears left-aligned, when it should be centered. I would wait to merge this until the hexString fix is ready, or add a workaround.

Other than that, :shipit:

**Comment by frosty** [comment]
@koke Turns out that CSS `text-align` was getting overridden by the paragraph style I was having to set on the text to remove padding. I've therefore switched to using a paragraph style to set the alignment of the text. I've also pulled in the latest from develop, which now includes a fix for `hexString` working with grayscale colors. 

Would you like to / would you mind giving this one more check over? Thank you!

**Comment by koke** [comment]
:shipit:

**Pull request merged**